core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/io/AwsChunkedEncodingInputStream.java

```diff
@@ -17,32 +17,42 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
-import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.core.checksums.SdkChecksum;
 import software.amazon.awssdk.core.internal.chunked.AwsChunkedEncodingConfig;
 import software.amazon.awssdk.core.io.SdkInputStream;
 import software.amazon.awssdk.utils.Logger;
+import software.amazon.awssdk.utils.Validate;
 
 /**
- * A wrapper of InputStream that implements chunked encoding.
- * <p/>
- * Each chunk will be buffered.
- * <p/>
+ * A wrapper of InputStream that implements pseudo-chunked-encoding.
+ * Each chunk will be buffered for the calculation of the chunk signature
+ * which is added at the head of each chunk.<br>
+ * The default chunk size cannot be customized, since we need to calculate
+ * the expected encoded stream length before reading the wrapped stream.<br>
  * This class will use the mark() & reset() of the wrapped InputStream if they
  * are supported, otherwise it will create a buffer for bytes read from
  * the wrapped stream.
  */
-@SdkInternalApi
+@SdkProtectedApi
```

Comment:
Should this still be in `software.amazon.awssdk.core.internal`?

Same feedback for any other classes that were escalated from `@SdkInternalApi` to `@SdkProtectedApi`.

Reply:
Good catch, this class is extended and used only by internal classes, this should be reverted back to @SdkInternalApi
as it was before https://github.com/aws/aws-sdk-java-v2/pull/2780